### PR TITLE
docs(observability): create observability dir and add log forwarding example

### LINKED ISSUE #1898: add a log forwarding example to docs
**Please describe what should be documented
Add how to forward logs to a sink via fluent bit in osm (log sink example can be Azure log analytics)

**Please suggest where in the repo the document should be located
docs/patterns/observability/log-forwarding.md

cc @SanyaKochhar

### PR DESCRIPTION
**Pull request by SanyaKochhar**
**Description**: Restructures metrics and logs documentation into a new observability directory and adds documentation for an example of log forwarding. Resolves #1898 

**Affected area**:

- New Functionality      [ ]
- Documentation          [x]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
Yes, Fluent Bit forwarding to Azure example is inspired by the example [here](https://samcogan.com/export-kubernetes-logs-to-azure-log-analytics-with-fluent-bit/)

### CODE REVIEW

File: docs/patterns/observability/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Observability
+
+OSM's observability stack includes Prometheus for metrics collection, Grafana for metrics visualization and Fluent Bit for log forwarding to a user-defined endpoint. These are disabled by default but may be enabled during OSM installation with flags `--enable-prometheus`, `--enable-grafana` and `--enable-fluentbit`. Alternatively, their values may be set in `.env`.
```
Comment (eduser25) [inline docs/patterns/observability/README.md:3]: > Alternatively, their values may be set in `.env`.

Is this still for install? not seeing env values grabbed at install time on install.go (cli).

Comment (SanyaKochhar) [reply docs/patterns/observability/README.md:3]: oh you're right, I think I got confused and that's true only for the demo. Removing now!

File: docs/patterns/observability/logs.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+By default, the Fluent Bit sidecar is configured to simply send logs to the Fluent Bit container's stdout. If you have installed OSM with Fluent Bit enabled, you may access these logs using `kubectl logs -n osm-system <osm-controller-name> -c fluentbit-logger`. However, we recommend configuring log forwarding to your preferred output for more informative results.
+
+You may configure log forwarding to an output by following these steps _before_ you install OSM.
+
+1. Define the output plugin you would like to forward your logs to in the existing `fluentbit-configmap.yaml` file by replacing the `[OUTPUT]` section with your chosen output as described by Fluent Bit documentation [here](https://docs.fluentbit.io/manual/v/1.4/pipeline/outputs).
+
+2. The existing configuration assumes OSM is running on a kind cluster so it uses a parser that matches the CRI log format. If you are using a kubernetes distribution that causes your logs to be formatted differently, you may need to update the `[PARSER]` section and the `parser` name in the `[INPUT]` section to one of the parsers defined [here](https://github.com/fluent/fluent-bit/blob/master/conf/parsers.conf).
```
Comment (eduser25) [inline docs/patterns/observability/logs.md:17]: Could the Kind part be obviated here, and just say it uses CRI log format parsing by default?

**Review by snehachhabria — DISMISSED** [review]

**Review by snehachhabria — APPROVED** [review]

**Review by eduser25 — APPROVED** [review]

**Review by eduser25 — APPROVED** [review]
lgtm

**Pull request merged**